tools/kmd_hmdb_data_plot/macro.xml

```diff
@@ -0,0 +1,165 @@
+      <option value="M+2ACN+H">M+2ACN+H</option>
+      <option value="2M+H">2M+H</option>
+      <option value="2M+NH4">2M+NH4</option>
+      <option value="2M+Na">2M+Na</option>
+      <option value="2M+K">2M+K</option>
+      <option value="M-H">M-H</option>
+      <option value="M-2H">M-2H</option>
+      <option value="M-H2O-H">M-H2O-H</option>
+      <option value="M+Cl">M+Cl</option>
+      <option value="M+FA-H">M+FA-H</option>
+      <option value="M+Hac-H">M+Hac-H</option>
+      <option value="M-H+HCOONa">M-H+HCOONa</option>
+      <option value="M+Br">M+Br</option>
+      <option value="M+TFA-H">M+TFA-H</option>
+      <option value="2M-H">2M-H</option>
+      <option value="2M+FA-H">2M+FA-H</option>
+      <option value="2M+Hac-H">2M+Hac-H</option>
+    </param>
+  </xml>
+
+  <xml name="produce_plot_inputs">
+    <param name="csv_input" type="data" format="tabular"
+      help="Tabular file to use to produce the plot."
+      label="A Tabular Input File"
+    />
+    <param
+      name="annotation_columns"
+      type="data_column"
+      data_ref="csv_input"
+      use_header_names="true"
+      multiple="true"
+      optional="true"
+      help="Select columns to show when a point of the graph is hovered"
+      label="Annotation columns"
+    />
+    <param
+      name="x_columns"
+      type="data_column"
+      data_ref="csv_input"
+      use_header_names="true"
+      optional="true"
+      multiple="true"
+      help="
+        Select one or multiple column to use as X values
+        to generate the graph. See the help section to better
+        understand the usage of multiple values
+      "
+      label="X Axis"
+    />
+    <param
+      name="y_columns"
+      type="data_column"
+      data_ref="csv_input"
+      use_header_names="true"
+      optional="true"
+      multiple="true"
+      help="
+        Select one or multiple column to use as Y values
+        to generate the graph. See the help section to better
+        understand the usage of multiple values
+      "
+      label="Y Axis"
+    />
+  </xml>
+
+  <xml name="not_get_data">
+    <param name="mz_ratio" value="unknown" type="hidden" />
+    <param name="mass_tolerance" value="unknown" type="hidden" />
+    <param name="database" value="unknown" type="hidden" />
+    <param name="adducts" value="unknown" type="hidden" />
+  </xml>
+
+  <xml name="not_produce_plot">
+    <param name="csv_input" value="unknown" type="hidden" />
+    <param name="annotation_columns" type="hidden" />
+    <param name="x_columns" type="hidden" />
+    <param name="y_columns" type="hidden" />
+  </xml>
+
+  <xml name="get_data_outputs">
+    <data name="output_path" format="csv"
+      label="csv - ${tool.name} on ${what.mz_ratio}±${what.mass_tolerance} - ${what.database}"
+    >
+      <filter>"get_data" in str(what['to_do'])</filter>
+    </data>
+  </xml>
+
+  <xml name="produce_plot_outputs">
+    <data name="output" format="html"
+      label="html on ${
```

Comment:
I would not do those acrobatics on output names. Those informations are all captured inside Galaxy and you can look at them via the GUI.